components/admin_console/plugin_management/plugin_management.jsx

```diff
@@ -967,8 +989,8 @@ export default class PluginManagement extends AdminSettings {
                             <div className='col-sm-8'>
                                 <div className='file__upload'>
                                     <button
-                                        className={uploadBtnClass}
-                                        disabled={!enableUploads || !enable}
+                                        className={classNames(['btn', {'btn-primary': enableUploads}])}
+                                        disabled={!enableUploads || !this.state.enable || this.state.requirePluginSignature}
```

Comment:
What's the difference between `enable` and `this.state.enable` and why do we use one over the other?

Reply:
`enable` is derived from `props` and doesn't update when user toggles this field. `this.state.enable` stores what the user sets locally and eventually is pushed to the server when `save` button is clicked. Reason we use `this.state.enable` is to enable/disable other fields when this field is set or not set, we can't do that when we use `this.props.enable` as it'll always be static till the server pulls the new data from the server.